transaction/transaction.go

```diff
@@ -358,9 +366,187 @@ func MakeAssetConfigTxn(account string, feePerByte, firstRound, lastRound uint64
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	tx, err := MakeAssetTransferTxn(account, recipient, "", amount,
+		feePerByte, firstRound, lastRound, note, genesisID, genesisHash, creator, index)
+
+	targetAddr, err := types.DecodeAddress(target)
+	if err != nil {
+		return tx, err
+	}
+	tx.AssetSender = targetAddr
+
+	// Update fee
+	eSize, err := estimateSize(tx)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+
+	tx.Fee = types.MicroAlgos(eSize * feePerByte)
+	if tx.Fee < MinTxnFee {
+		tx.Fee = MinTxnFee
+	}
+
+	return tx, err
+}
+
+// MakeAssetMintTransaction creates a tx for minting more of an asset (transferring from the reserve account)
+// Note that this is just an asset transfer transaction where sender=reserve
+// - reserve is a checksummed, human-readable address that will send the transaction. it must be the asset's reserve to be considered a minting, but this is not enforced by the function.
+// - recipient is a checksummed, human-readable address; it will receive the revoked assets
+// - amount is the number of assets to mint
+// - feePerByte is a fee per byte
+// - firstRound is the first round this txn is valid (txn semantics unrelated to asset management)
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetMintTxn(reserve, recipient string, amount, feePerByte, firstRound, lastRound uint64, note []byte,
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	tx, err := MakeAssetTransferTxn(reserve, recipient, "", amount,
+		feePerByte, firstRound, lastRound, note, genesisID, genesisHash, creator, index)
+	return tx, err
+}
+
+// MakeAssetBurnTransaction creates for burning an asset (transferring to the reserve account)
+// Note that this is just an asset transfer transaction where recipient=reserve
+// - account is a checksummed, human-readable address that will send the transaction.
+// - reserve is a checksummed, human-readable address; it will receive the revoked assets and must be the reserve account to be considered a burn. this is not enforced by the function
+// - amount is the number of assets to "burn"
+// - feePerByte is a fee per byte
+// - firstRound is the first round this txn is valid (txn semantics unrelated to asset management)
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetBurnTxn(account, reserve string, amount, feePerByte, firstRound, lastRound uint64, note []byte,
```

Comment:
Same here. The rational behind removing these functions is that they no real added value. If the functions were to compute or pull the relevant information automatically (such as the reserve address and so on), I'd agree, but right now, I think it may be more confusing to the user. Again, worth discussion.